src/components/fx/hoverlabel_defaults.js

```diff
@@ -9,10 +9,32 @@
 'use strict';
 
 var Lib = require('../../lib');
+var isUnifiedHover = require('./helpers').isUnifiedHover;
 
 module.exports = function handleHoverLabelDefaults(contIn, contOut, coerce, opts) {
     opts = opts || {};
 
+    function inheritFontAttr(attr) {
+        if(!opts.font[attr]) {
+            if(contIn.legend && contIn.legend.font && contIn.legend.font[attr]) {
```

Comment:
is `contOut.legend` not available yet? Could we rearrange so it *is* available? As it is this won't work with templates.

Reply:
> is `contOut.legend` not available yet? Could we rearrange so it _is_ available?

It's not available yet. I'm not sure how difficult it would be to rearrange so it is. I'm not super familiar with the `supplyDefaults` routine but it does look like the legend (which depend on data) is coerced in the final steps:
https://github.com/plotly/plotly.js/blob/7c67896603d6c61d515a5fb88d653580bd4e748a/src/plots/plots.js#L440-L441

Reply:
Done in https://github.com/plotly/plotly.js/pull/4687/commits/dfd73b86ab1ebcf8448245583035396a43c71b07 which results in simpler code (in particular I don't have to check for the existence of nested keys in `legend` object: if it's been coerced, everything is there)!